### PR TITLE
Feature/1124/mark active

### PR DESCRIPTION
**Pull request by cjhaddad**
# 💅🏼 What issue does this fix?
Implements #1124 #1141 for mark inactive functionality. Still WIP until tests are fixed/ written and interactions with API are fixed/ working. 

## 🍂 Before You Submit

* [x] Did you write tests?
* [ ] Did you run Google Lighthouse and/or WebAIM (Wave) on UI components in your PR?
* [ ] Does your PR contain any required translations?
* [ ] Did you run `bundle exec rspec` from the root?
* [ ] Did you run `bundle exec rails rswag` from the root?
* [ ] Did you run `bundle exec rubocop` from the root?
* [x] Did you run `yarn lint` in `/client`?
* [x] Did you run `yarn test` in `/client`?
* [ ] Are your primary keys UUIDs on any new tables?

## 🛷 Deployment

* [ ] Data Migrations
* [ ] Schema Migrations
* [ ] Dependencies

## 🧳 Steps to test
This ticket is to test both  #1124 #1141. There should now be an 'Actions' column in both Nebraska and Illinois dashboard tables. The column should be the last one. For each row there is a button that reads "Mark Inactive." Upon clicking this button a modal shows up that allows you to select the reason for marking inactive and the inactive date. (see figma for specific UI/ desgin: https://www.figma.com/file/uCveVQQnYcw3MsenD1PNcG/P4P-2.0?node-id=4979%3A23572) When you select "Mark Inactive" in the modal a call is then made to the backend sending this information through and updating the specific child. That child row is then dropped to the bottom of the table and all columns for that row are replaced with a dash, expect for the Actions cell the Mark Inactive button is grayed out, and there is now 'Inactive' next to the child's name.

### CODE REVIEW

**Comment by katelovescode** [comment]
Is there any way to control the width of the dropdown?  When you've picked "Other" and try to pick something else, you can't see enough of the text to determine what the other options are.

Screen Shot 2021-05-11 at 9 27 24 AM

@rebeccakarasiktw do we want the "mark inactive" to be removed from inactive kids?  It might be confusing if people think they can click it to reactivate a kid

Screen Shot 2021-05-11 at 9 27 51 AM

**Review by katelovescode — DISMISSED** [review]
My comments are almost all about "should we do something like x?" because I haven't been in the UI conversations very in-depth, so please feel free to push that scope out to another ticket.  The only thing I see that actually needs changing are the typos on `receiving` - I'm ✅ after those are fixed!

File: client/src/Dashboard/DashboardTable.js [hunk 1]
```diff
@@ -260,30 +366,106 @@ export default function DashboardTable({ tableData, userState, setActiveKey }) {
...
+        <Select
+          className="inactive-select"
+          placeholder="Reason for making inactive"
+          bordered={false}
+          onChange={value => setInactiveReason(value)}
+        >
+          <Select.Option value="no_longer_in_my_care">
+            {t('inactiveReason1')}
+          </Select.Option>
+          <Select.Option value="no_longer_recieving_subsidies">
```
[the remark this answers is not in the document]
Comment (cjhaddad) [reply client/src/Dashboard/DashboardTable.js:449]: 😂 my bad spelling

**Comment by cjhaddad** [comment]
> Is there any way to control the width of the dropdown? When you've picked "Other" and try to pick something else, you can't see enough of the text to determine what the other options are.
>  re: this I will look into it! I am quite sure it can be adjusted. 
> Screen Shot 2021-05-11 at 9 27 24 AM
> 
> @rebeccakarasiktw do we want the "mark inactive" to be removed from inactive kids? It might be confusing if people think they can click it to reactivate a kid

> Screen Shot 2021-05-11 at 9 27 51 AM

**Comment by cjhaddad** [comment]
> This might be out of the scope of this PR, so feel free to move it to another ticket if so, but should inactive kids stay at the bottom during sorting? Or be "grouped" (i.e. the active kids a-z, then the inactive kids a-z)? I don't know if this is hard or even the right UX, just a thought I had cc @rebeccakarasiktw
> 
> sorting

Rebecca and I chatted about this, and she thought this behavior is ok -- allowing the user to ultimately be able to sort all children regardless of active status.

**Review by katelovescode — APPROVED** [review]

**Pull request merged**